presto-kafka07/src/main/java/com/facebook/presto/kafka/KafkaMetadata.java

```
@@ -225,6 +227,11 @@ public ColumnMetadata getColumnMetadata(ConnectorSession session, ConnectorTable
         }
 
         log.info("startTs: %s, endTs: %s", startTs, endTs);
+        if (config.isHardLimitOn() && startTs == null && endTs == null) {
```

I don't understand how the hard_limit is applied. Can you add more comments to make it clear? Especially for the  `for (Map.Entry<ColumnHandle, Domain> entry : columnHandleDomainMap.entrySet())` loop above.

Also, could we check whether it meets the expectations:
1. If users specify a time range exceeding the limitation, an exception should be thrown directly. Otherwise, the behavior is changed silently and unexpectedly.
2. If users do not specify the range, we could either throw an exception directly, or trim to the limited time range with an **explicit logging/print-out** notifying users.

---

Here is some thoughts about your expectations:
1. If the user doesn't know how to specify a range, they will fall into default query interval.
2. If the user knows how to specify a range, they must agree with default query interval and must have a specific reason to go beyond the default query interval. It would be costly to serve the users if they actually have this specific request. It will involve code review and service deploy processes which wouldn't the ideal way for serving our customers for urgent needs.

---

1. If the user doesn't know how to specify a range, Presto should either throw an exception **directly**, or trim to the limited time range with an **explicit logging/print-out notifying** users. Siliently unexpected behavior is **bad**.

2. A design should be resilient to users' mistakes/errors, rather than to rely on users always providing safe input. Also, this is what **limitation** means: it should not happen.

If users want to query exceeding the limitation, they need to notify all stakeholders, for instance, messaing team, since it could potentially break the whole kafka service, leading to a SEV0 or SEV1.